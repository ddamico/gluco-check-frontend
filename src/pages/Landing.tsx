import React from "react";
import { useTranslation } from "react-i18next";
import { Container, Grid, makeStyles, Typography } from "@material-ui/core";

import * as firebase from "firebase/app";
import * as firebaseui from "firebaseui";
import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
import { auth } from "../lib/firebase";

const assistantIcon = require("../images/icon-assistant.svg");

const useStyles = makeStyles((theme) => ({
  container: {
    paddingTop: theme.spacing(2),
    textAlign: "center",
  },
  contentGrid: {
    height: "90vh",
  },
  utterance: {
    fontWeight: "bold",
  },
  callResponseContainer: {
    "& p": {
      marginBottom: theme.spacing(2),
    },
    display: "inline-block",
    "& img": {
      width: "54px",
      height: "54px",
      display: "block",
      paddingBottom: theme.spacing(2),
    },
    "& .assistantBubble-response": {
      padding: "14px 24px",
      border: "2px solid #e8eaed",
      borderRadius: "62px",
      display: "inline-block",
    },
    [theme.breakpoints.down("sm")]: {
      "& .assistantBubble-response": {
        fontSize: ".9rem",
      },
    },
  },
}));

function Landing() {
  const classes = useStyles();
  const { t } = useTranslation();

  const firebaseUIConfig = {
    credentialHelper: firebaseui.auth.CredentialHelper.NONE,
    signInFlow: "redirect",
    signInSuccessUrl: "/",
    signInOptions: [
      {
        provider: firebase.auth.GoogleAuthProvider.PROVIDER_ID,
        fullLabel: "Let's get started",
      },
    ],
  };

  return (
    <Grid
      className={classes.contentGrid}
      container
      direction="column"
      justify="center"
      alignItems="center"
      spacing={2}
    >
      <Grid item>
        <Container maxWidth="md">
          <div className={classes.callResponseContainer}>
            <Typography variant="h5" component="p">
              Hey Google, ask Gluco Check my blood sugar
            </Typography>
            <div>
              <img src={assistantIcon} alt=""></img>
              <Typography
                variant="h5"
                component="p"
                className="assistantBubble-response"
                noWrap
              >
                6.5 and steady as of five minutes ago
              </Typography>
            </div>
          </div>
        </Container>
      </Grid>
      <Grid item>
        <Container maxWidth="sm">
          <Typography variant="body1">
            {t("landing.introduction.p1")}{" "}
            <span className={classes.utterance}>
              Hey Google, ask Gluco Check my BG
            </span>
            . Or{" "}
            <span className={classes.utterance}>
              Hey Google, ask Gluco Check my sensor's age
            </span>
            . Or{" "}
            <span className={classes.utterance}>
              Hey Google, ask Gluco Check my IOB
            </span>
            .
          </Typography>
        </Container>
      </Grid>
      <Grid item>
        <StyledFirebaseAuth uiConfig={firebaseUIConfig} firebaseAuth={auth} />
      </Grid>
    </Grid>
  );
}

export default Landing;
