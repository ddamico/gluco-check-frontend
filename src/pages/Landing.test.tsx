import React from "react";
import { cleanup, render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import Landing from "./Landing";

expect.extend(toHaveNoViolations);

afterEach(cleanup);

describe("Landing page", () => {
  it("renders the component", () => {
    const { container } = render(<Landing />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    const { container } = render(<Landing />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
