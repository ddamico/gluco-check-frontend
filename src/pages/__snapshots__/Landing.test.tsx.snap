// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page renders the component 1`] = `
<div
  class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthXl"
>
  <h2
    class="MuiTypography-root MuiTypography-h6"
  >
    landing.title
  </h2>
  <p>
    landing.introduction.p1
  </p>
  <ul>
    <li>
      landing.introduction.features1
    </li>
    <li>
      landing.introduction.features2
    </li>
    <li>
      landing.introduction.features3
    </li>
    <li>
      landing.introduction.features4
    </li>
    <li>
      landing.introduction.features5
    </li>
    <li>
      landing.introduction.features6
    </li>
    <li>
      landing.introduction.features7
    </li>
  </ul>
</div>
`;
